src/bin/tree-sitter-graph/main.rs

```diff
@@ -48,6 +48,13 @@ fn main() -> Result<()> {
         )
         .arg(Arg::with_name("scope").long("scope").takes_value(true))
         .arg(Arg::with_name("json").long("json").takes_value(false))
+        .arg(
+            Arg::with_name("output")
+                .short('o')
+                .long("output")
+                .requires("json")
+                .takes_value(true),
+        )
```

Comment:
This was (mercifully) all I needed to allow `-o`/`--output` to be available only when `--json` has been used.